File: 백준/Gold/10986. 나머지 합/나머지 합.py
import sys

I = sys.stdin.readline

answer = 0
n, m = map(int, I().split())
nums = list(map(int, I().split()))
s_nums = [0] * n
counter = [0] * m
s_nums[0] = nums[0]

for i in range(1, n):
    s_nums[i] = s_nums[i - 1] + nums[i]

for i in range(n):
    remainder = s_nums[i] % m
    if remainder == 0:
        answer += 1
    counter[remainder] += 1

for i in range(m):
    if counter[i] > 1:
        answer += counter[i] * (counter[i] - 1) // 2

print(answer)

File: 백준/Gold/10986. 나머지 합/나머지 합.js
const fs = require('fs');
const input = fs.readFileSync('./dev/stdin').toString().split('\n');

const [N, M] = input[0].split(' ').map(Number);
const A = input[1].split(' ').map(Number);
const prefixSum = [];
const counter = {
  0: 1,
};

for (let i = 0; i < N; i++) {
  if (i === 0) prefixSum.push(A[i]);
  else prefixSum.push(prefixSum[i - 1] + A[i]);
}

for (let i = 0; i < N; i++) {
  let processed = prefixSum[i] % M;
  if (processed in counter) counter[processed] += 1;
  else counter[processed] = 1;
}

let result = 0;

for (let i = 0; i < M; i++) {
  if (i in counter) result += parseInt((counter[i] * (counter[i] - 1)) / 2);
}

console.log(result);
